cardano-db/src/Cardano/Db/Migration/Version.hs

```diff
@@ -23,9 +23,9 @@ data MigrationVersion = MigrationVersion
   } deriving (Eq, Ord, Show)
 
 
-parseMigrationVersionFromFile :: String -> Maybe MigrationVersion
+parseMigrationVersionFromFile :: Text -> Maybe MigrationVersion
```

Comment:
Why are these types changing from `String` to `Text`? It does not seem to be related to "Modify cardano-sync to enable reusing from SMASH:.

Reply:
Well. 
First, I noticed that it would be better if I add `cardano-prelude` in a single module.
Then, there was a collision in that module between `cardano-prelude` and base Prelude.
Then I added the extension for the whole library which I consider worthy, `NoImplicitPrelude`.
Then other modules started complaining.
Then I fixed the other modules.

After a while, I realized it might have been better to do this in another PR, but you don't know how extensive the changes are until you do them.